static/src/javascripts/projects/common/modules/commercial/user-features.ts

```diff
@@ -35,124 +39,167 @@ const CONTRIBUTIONS_REMINDER_SIGNED_UP = {
+		value: String(JsonResponse.contentAccess.digitalPack),
+	});
+	setCookie({
+		name: HIDE_SUPPORT_MESSAGING_COOKIE,
+		value: String(!JsonResponse.showSupportMessaging),
+	});
 	if (JsonResponse.oneOffContributionDate) {
-		addCookie(
-			ONE_OFF_CONTRIBUTION_DATE_COOKIE,
-			JsonResponse.oneOffContributionDate,
-		);
+		setCookie({
+			name: ONE_OFF_CONTRIBUTION_DATE_COOKIE,
+			value: JsonResponse.oneOffContributionDate,
+		});
 	}
 
-	removeCookie(ACTION_REQUIRED_FOR_COOKIE);
-	if ('alertAvailableFor' in JsonResponse) {
-		addCookie(ACTION_REQUIRED_FOR_COOKIE, JsonResponse.alertAvailableFor);
+	removeCookie({ name: ACTION_REQUIRED_FOR_COOKIE });
+	if (JsonResponse.alertAvailableFor) {
+		setCookie({
+			name: ACTION_REQUIRED_FOR_COOKIE,
+			value: JsonResponse.alertAvailableFor,
+		});
 	}
 
 	if (
 		adFreeDataIsPresent() &&
 		!forcedAdFreeMode &&
 		!JsonResponse.contentAccess.digitalPack
 	) {
-		removeCookie(AD_FREE_USER_COOKIE);
+		removeCookie({ name: AD_FREE_USER_COOKIE });
 	}
 
 	if (JsonResponse.contentAccess.digitalPack) {
-		addCookie(AD_FREE_USER_COOKIE, timeInDaysFromNow(2));
+		setCookie({ name: AD_FREE_USER_COOKIE, value: timeInDaysFromNow(2) });
 	}
 };
 
-const deleteOldData = () => {
+const deleteOldData = (): void => {
 	// We expect adfree cookies to be cleaned up by the logout process, but what if the user's login simply times out?
-	removeCookie(USER_FEATURES_EXPIRY_COOKIE);
-	removeCookie(PAYING_MEMBER_COOKIE);
-	removeCookie(RECURRING_CONTRIBUTOR_COOKIE);
-	removeCookie(AD_FREE_USER_COOKIE);
-	removeCookie(ACTION_REQUIRED_FOR_COOKIE);
-	removeCookie(DIGITAL_SUBSCRIBER_COOKIE);
-	removeCookie(HIDE_SUPPORT_MESSAGING_COOKIE);
-	removeCookie(ONE_OFF_CONTRIBUTION_DATE_COOKIE);
+	removeCookie({ name: USER_FEATURES_EXPIRY_COOKIE });
+	removeCookie({ name: PAYING_MEMBER_COOKIE });
+	removeCookie({ name: RECURRING_CONTRIBUTOR_COOKIE });
+	removeCookie({ name: AD_FREE_USER_COOKIE });
+	removeCookie({ name: ACTION_REQUIRED_FOR_COOKIE });
+	removeCookie({ name: DIGITAL_SUBSCRIBER_COOKIE });
+	removeCookie({ name: HIDE_SUPPORT_MESSAGING_COOKIE });
+	removeCookie({ name: ONE_OFF_CONTRIBUTION_DATE_COOKIE });
 };
 
 const requestNewData = () =>
-	fetchJson(`${config.get('page.userAttributesApiUrl')}/me`, {
-		mode: 'cors',
-		credentials: 'include',
-	})
+	fetchJson(
+		`${config.get<string>(
+			'page.userAttributesApiUrl',
+			'/USER_ATTRIBUTE_API_NOT_FOUND',
+		)}/me`,
+		{
+			mode: 'cors',
+			credentials: 'include',
+		},
+	)
+		.then((response) => {
+			if (!validateResponse(response))
+				throw new Error('invalid response');
+			return response;
+		})
 		.then(persistResponse)
-		.catch(() => {});
+		.catch(noop);
 
-const datedCookieIsOld = (datedCookieName) => {
-	const expiryDateFromCookie = getCookie(datedCookieName);
+const datedCookieIsOldOrMissing = (datedCookieName: string): boolean => {
+	const expiryDateFromCookie = getCookie({ name: datedCookieName });
+	if (!expiryDateFromCookie) return true;
```

Comment:
Minor non-blocking - just some naming improvements if you agree:
```suggestion
const cookieIsExpiredOrMissing = (cookieName: string): boolean => {
	const expiryDateFromCookie = getCookie({ name: cookieName });
	if (!expiryDateFromCookie) return true;
```